R3SOL-0  Remove hibernate dependency from ledger persistence lib

driessamyn opened the pull request:
Removes Hibernate dependency from the ledger-persistence lib.

Commit:
Example case using JDBC directly.

Commit:
Remove use of EntityManager from Repo

blsemo commented on libs/ledger-lib-persistence/src/main/kotlin/net/corda/ledger/libs/persistence/utxo/impl/UtxoPersistenceServiceImpl.kt:
```diff
@@ -173,11 +180,15 @@ class UtxoPersistenceServiceImpl(
     ): Map<SecureHash, String> {
-        return entityManagerFactory.transaction { em ->
-            repository.findSignedTransactionIdsAndStatuses(em, transactionIds)
+        entityManagerFactory.createEntityManager().use { em ->
```
Why did you have to remove the transaction here?

Commit:
Move EMF out of lib

Commit:
detekt fix

Commit:
unit test fix

Commit:
detekt fix

Commit:
Remove hibernate and db-orm dependencies

Commit:
Remove debug print statements.

Commit:
Detekt fixes.

Commit:
Attempt to fix the transaction management in the `persistTransactionIfDoesNotExist` case.

NOTE: includes a lot of println lines for debugging, will be removed once verified.

Commit:
detekt fix.

Commit:
Remove debugging query that fails in unit test

Commit:
Remove printline debug statements.
